icu4c/source/i18n/collationruleparser.cpp

```diff
@@ -622,7 +622,7 @@ CollationRuleParser::parseSetting(UErrorCode &errorCode) {
                 setParseError("expected language tag in [import langTag]", errorCode);
                 return;
             }
-            if(length == 3 && uprv_memcmp(baseID, "und", 3) == 0) {
+            if(length == 0 || (length == 3 && uprv_memcmp(baseID, "und", 3) == 0)) {
```

Comment:
This has always been missing a check for a separator/terminator following after "und", hasn't it?

It does not actually look right: If the whole baseID is "und", then loading from "root" is good, but if it's "und_IN" == "_IN", we should load from that.